src/zarr/testing/store.py

```diff
@@ -1,61 +1,121 @@
+from typing import Generic, TypeVar
+
 import pytest
 
 from zarr.abc.store import Store
 from zarr.buffer import Buffer
+from zarr.store.core import _normalize_interval_index
 from zarr.testing.utils import assert_bytes_equal
 
+S = TypeVar("S", bound=Store)
+
+
+class StoreTests(Generic[S]):
+    store_cls: type[S]
```

Comment:
`StoreTests` is now parametrized by the type of the store being tested, because I am lazy and I wanted my IDE to automatically fill out the type annotations for various methods of this class.